src/track/track.cpp

```diff
@@ -290,6 +299,10 @@ bool Track::replaceRecord(
         bpmUpdatedFlag = trySetBpmWhileLocked(
                 newRecord.getMetadata().getTrackInfo().getBpm().value());
     }
+
+    if (forceBpm) {
+        setBpmLocked(wasLocked);
```

Comment:
Again, this is not permissible. The object is in an inconsistent state and locked.